### PR TITLE
Refs Task#385488 Unit test for update Bot

### PR DESCRIPTION
**Pull request by tamtt-1031**
## Related Tickets

- [#385488](https://dev.sun-asterisk.com/issues/385488)

## What's this PR do ?

- [x] Unit test for update Bot

## Library
*(List package, library third party add new include version)*

- N/A

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr></table>

File: app/Http/Requests/BotUpdateRequest.php [hunk 1]
```diff
@@ -26,18 +26,27 @@ public function authorize()
...
-            'bot_key' => 'required|max:50',
-            'bot_key' => Rule::unique('bots')->ignore($this->id)->where(function ($query) {
-                return $query->where('user_id', \Auth::id());
-            }),
+            'name' => [
+                'required',
+                'max:50',
+                Rule::unique('bots')->ignore($this->id)->where(function ($query) {
+                    return $query->where('user_id', \Auth::id());
+                })
```
Comment (framgiaciv3) [inline app/Http/Requests/BotUpdateRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Http/Requests/BotUpdateRequest.php</br>**Line**: 34</br>Comma required after last value in array declaration

File: app/Http/Requests/BotUpdateRequest.php [hunk 2]
```diff
@@ -26,18 +26,27 @@ public function authorize()
...
+            ],
+            'cw_id' => [
+                'required',
+                'max:10',
+                Rule::unique('bots')->ignore($this->id)->where(function ($query) {
+                    return $query->where('user_id', \Auth::id());
+                })
```
Comment (framgiaciv3) [inline app/Http/Requests/BotUpdateRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Http/Requests/BotUpdateRequest.php</br>**Line**: 41</br>Comma required after last value in array declaration

File: app/Http/Requests/BotUpdateRequest.php [hunk 3]
```diff
@@ -26,18 +26,27 @@ public function authorize()
...
+            ],
+            'bot_key' => [
```
Comment (framgiaciv3) [inline app/Http/Requests/BotUpdateRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Http/Requests/BotUpdateRequest.php</br>**Line**: 43</br>Each line in an array declaration must end in a comma

File: app/Http/Requests/BotUpdateRequest.php [hunk 4]
```diff
@@ -26,18 +26,27 @@ public function authorize()
...
+                'required',
+                'max:50',
+                Rule::unique('bots')->ignore($this->id)->where(function ($query) {
+                    return $query->where('user_id', \Auth::id());
+                })
```
Comment (framgiaciv3) [inline app/Http/Requests/BotUpdateRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Http/Requests/BotUpdateRequest.php</br>**Line**: 48</br>Comma required after last value in array declaration

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr></table>

File: tests/Feature/ChatbotControllerTest.php [hunk 5]
```diff
@@ -372,19 +372,253 @@ public function testEditBotPermissionDenine()
...
+            'name' => 'Updated Bot',
+            'cw_id' => '123456789',
+            'bot_key' => 'asdg12asd3423adasdasd23sdasdas23',
+        ];
+        
         $this->actingAs($user);
-        $response = $this->put(route('bots.update', $bot->id), ['name' => 'New Name']);
+        $response = $this->put(route('bots.update', $bot->id), $params);
+        $response->assertRedirect('bots/' . $bot->id . '/edit');
+        $this->assertEquals('Updated Bot', Bot::first()->name);
```
Comment (ghost) [inline tests/Feature/ChatbotControllerTest.php]: $bot = Bot::first();
rồi phía dưới gọi dùng lại cho đở query a
ko a viết thành thế này nhìn đẹp hơn này
```
$this->assertDatabaseHas('bots', ['id' => $bot->id, 'name' => 'Updated Bot', 'cw_id' => '123456789', 'bot_key' => ..])
```

**Review by ghost — APPROVED** [review]

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr></table>

**Review by nguyenthinhhn — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**